modules/exploits/multi/http/coldfusion_ckeditor_file_upload.rb

```diff
@@ -0,0 +1,106 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  include Msf::Exploit::Remote::HttpClient
+
+  Rank = ExcellentRanking
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'            => 'Adobe ColdFusion CKEditor unrestricted file upload',
+      'Description'     => %q{
+        A file upload vulnerability in the CKEditor of Adobe ColdFusion 11
+        (Update 14 and earlier), ColdFusion 2016 (Update 6 and earlier), and
+        ColdFusion 2018 (July 12 release) allows unauthenticated remote
+        attackers to upload and execute JSP files through the filemanager
+        plugin.
+        Tested on Adobe ColdFusion 2018.
+      },
+      'Author'          =>
+        [
+          'Pete Freitag de Foundeo', # Vulnerability discovery
+          'Vahagn vah_13 Vardanian', # First public PoC
+          'Qazeer' # Metasploit module
+        ],
+      'License'         => MSF_LICENSE,
+      'References'      =>
+        [
+          [ "CVE", "2018-15961" ],
+          [ "BID", "105314" ],
+          [ "URL", "https://helpx.adobe.com/fr/security/products/coldfusion/apsb18-33.html" ]
+        ],
+      'Privileged'      => false,
+      'Stance'          => Msf::Exploit::Stance::Aggressive,
+      'Platform'        => ['win', 'linux'],
+      'Targets'        =>
```

Comment:
The Java payload is platform agnostic, so you should be able to forgo having separate targets, and instead do this:

```ruby
          [ 'Java Universal',
            {
                'Arch' => ARCH_JAVA,
                'Platform' => 'java'
            },
          ]
```

Take a look at `grep -rn ".jsp" modules/exploits/multi/ | grep upload` for a bunch of examples.

If you want to support multiple targets, take a look at [struts_code_exec_exception_delegator](https://github.com/rapid7/metasploit-framework/tree/master/modules/exploits/multi/http/struts_code_exec_exception_delegator.rb) or [openfire_auth_bypass](https://github.com/rapid7/metasploit-framework/tree/master/modules/exploits/multi/http/openfire_auth_bypass.rb) as examples which supports the above universal target, plus a target for windows and a target for linux. They aren't JSP file upload exploits, but are similar in principle.